scripts/generate-import-linter-config.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python3
+import argparse
+import configparser
+import importlib
+import pkgutil
+from collections.abc import ItemsView
+from typing import Callable, Literal, cast
+
+from typing_extensions import NotRequired, TypedDict
+
+
+AlertingLevel = Literal["error", "warn", "none"]
+ValueType = AlertingLevel | str | list[str]
+
+
+class Contract(TypedDict):
+    name: str
+    type: Literal["layers", "independence", "forbidden"]
+    layers: NotRequired[list[str]]
+    modules: NotRequired[list[str]]
+    ignore_imports: NotRequired[list[str]]
+    unmatched_ignore_imports_alerting: NotRequired[AlertingLevel]
+    source_modules: NotRequired[list[str]]
+    forbidden_modules: NotRequired[list[str]]
+
+
+CONTRACTS = "importlinter:contracts"
+STATIC_CONFIG = {
+    "importlinter": {
+        "root_package": "ggshield",
+        "include_external_packages": True,
+    },
+    CONTRACTS: [
+        {
+            "name": "ggshield-layers",
+            "type": "layers",
+            "layers": [
+                "ggshield.cmd|*",
+                "ggshield.verticals|*",
+                "ggshield.core",
+                "ggshield.utils | pygitguardian",
+            ],
+            "ignore_imports": [
+                "ggshield.cmd.main -> ggshield.cmd.**",
+                "ggshield.cmd.** -> ggshield.cmd.utils.*",
+            ],
+            "unmatched_ignore_imports_alerting": "warn",
+        },
+        {
+            "name": "verticals-cmd-transversals",
+            "type": "forbidden",
+            "source_modules": [
+                "ggshield.cmd.*",
+            ],
+            "forbidden_modules": [
+                "ggshield.verticals.*",
+            ],
+            "ignore_imports": [
+                "ggshield.cmd.main -> ggshield.verticals.**",
+                "ggshield.cmd.{}.** -> ggshield.verticals.{}",
+                "ggshield.cmd.{}.** -> ggshield.verticals.{}.**",
+                # FIXME: #521 - enforce boundaries between cmd.auth and verticals.hmsl
+                "ggshield.cmd.auth.** -> ggshield.verticals.hmsl.**",
+            ],
+            "unmatched_ignore_imports_alerting": "none",
+        },
+    ],
+}
+
+
+def get_submodules(*, name: str, prefixed: bool) -> list[str]:
+    """Retrieve the modules inclued in a package"""
+    module = importlib.import_module(name)
+    return [
+        x.name
+        for x in pkgutil.iter_modules(
+            path=module.__path__,
+            prefix=f"{module.__name__}." if prefixed else "",
+        )
+    ]
+
+
+def expand_glob(line: str) -> list[str]:
+    """
+    Expand the *glob*
+
+    supported globs:
+    - "xxx.*" => ["xxx.sub_module_a", "xxx.sub_module_b"]
+    - "xxx|*" => ["xxx.sub_module_a | xxx.sub_module_b"]
+    - "xxx" (no globs) => ["xxx"]
```

Comment:
The glob syntax makes `STATIC_CONFIG` hard to read because some `*` are expanded and others are not. I'd prefer if the globs expanded by the script used a different character, for example `%` (like in SQL).

Reply:
This is a very good idea!